### PR TITLE
Morphology operators

### PR DESCRIPTION
**Pull request by cquammen**
Added a number of binary morphology operators. Note that this is a follow-on to PR #640.

### CODE REVIEW

**Comment by cquammen** [comment]
@cryos @cjh1 @ElliotPadgett This should be ready for review.

To test:
* Launch Tomviz
* Open small data set
* Crop data set to something like a 50x50x50 cube for quicker testing
* Run Binary Threshold
* Select the child data set produced by Binary Threshold
* Create a Slice plane to visualize the binary label map
* Run Binary Dilate, default parameters. Zoom in to the data (the Slice gets translated; separate bug). The segmented object will be enlarged.
* Delete Binary Dilate. Try the other morphology filters (Binary Erode, Binary Open, Binary Close)

**Comment by cjh1** [comment]
@cquammen I am happy to review this, but not sure I am using it right, an chance of doing a quick hangout to walk me through it?

File: tomviz/python/BinaryDilate.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        dilate_filter.Update()
+        itk_image_data = dilate_filter.GetOutput()
+
+        PyBufferType = itk.PyBuffer[itk_input_image_type]
+        label_buffer = PyBufferType.GetArrayFromImage(itk_image_data)
+
+        utils.set_array(dataset, label_buffer)
+    except Exception as exc:
+        print("Exception encountered while running BinaryDilate")
+        print(exc)
```
Comment (cjh1) [inline tomviz/python/BinaryDilate.py]: Show we re-raise these exceptions so we get a traceback ( once its fixed ). print(exc) with just print the exception type and message.

**Comment by cquammen** [comment]
Thanks for the review. I'll rebase and merge.

**Pull request merged**